.github/workflows/CI-python.yml

```diff
@@ -94,19 +93,13 @@ jobs:
         name: Run widget tests
         run: yarn e2e-widget
 
-      - if: ${{ matrix.packageDirectory == 'raiwidgets' }}
-        name: Run tests
-        run: |
-          pytest --durations=10
-        working-directory: ${{ matrix.packageDirectory }}
-
       - name: Upload code coverage results
         uses: actions/upload-artifact@v2
         with:
           name: ${{ matrix.packageDirectory }}-code-coverage-results
           path: ${{ matrix.packageDirectory }}/htmlcov
         # Use always() to always run this step to publish test results when there are test failures
-        if: ${{ always() && matrix.packageDirectory != 'raiwidgets'}}
+        if: ${{ always()}}
```

Comment:
This seems like it'll always evaluate to true so you can probably remove the line

Reply:
the difference is with "always" this step will always execute regardless whether previous steps successes or failed.
So we can always see the coverage even test failed.